internal/agent/accrual.go

```diff
@@ -0,0 +1,196 @@
+package agent
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/pinbrain/gophermart/internal/logger"
+	"github.com/pinbrain/gophermart/internal/model"
+)
+
+const (
+	// Интервал проверки наличия необработанных заказов
+	checkInterval = 10 * time.Second
+	// Количество горутин, отправляющих запросы в accrual
+	workerCount = 5
+)
+
+var ErrReqLimit = errors.New("too many requests")
+
+type Storage interface {
+	GetOrdersToProcess(ctx context.Context) ([]model.Order, error)
+	UpdateOrderStatus(ctx context.Context, orderID int, status model.OrderStatus, accrual float64) error
+}
+
+type AccrualAgent struct {
+	appCtx     context.Context
+	storage    Storage
+	accrualURL string
+
+	rateLimit        sync.RWMutex
+	rateLimitEndTime time.Time
+}
+
+func NewAccrualAgent(ctx context.Context, storage Storage, accrualURL string) AccrualAgent {
```

Comment:
```suggestion
func NewAccrualAgent(ctx context.Context, storage Storage, accrualURL string) *AccrualAgent {
```
Надо либо внутри мьютекс определять, как ссылку, либо агента возвращать в виде ссылки. Иначе при передаче этого объекта внутрь других структур или функций получим дублирование мьютекса и, как следствие, неработающие блокировки.